awssdk/src/it/scala/DeleteOpsSpec.scala

```diff
@@ -40,4 +40,49 @@ class DeleteOpsSpec extends ITSpec {
             delete >> Util.retryOf(get)(_.isEmpty)
       }.unsafeToFuture().futureValue shouldEqual None
   }
+
+  it should "success deleting item if key(s) exists by using condition expression" in forAll {
+    test: TestDataSimple =>
+      partitionKeyTable[IO].use {
+        case (client, table) =>
+          val put = client.put[TestDataSimple](table.tableName, test)
+          val delete = client.delete(
+            table,
+            test.id,
+            Expression(
+              "attribute_exists(#id)",
+              Map("#id" -> "id"),
+              Map.empty
+            )
+          )
+          val get = client.get[Id, TestDataSimple](
+            table,
+            test.id,
+            consistentRead = false
+          )
+          put >> Util.retryOf(get)(_.isDefined) >>
+            delete >> Util.retryOf(get)(_.isEmpty)
```

Comment:
I think this test can potentially be true positive if there is a bug implimentation. i.e. if `Expression` is not being passed to the Java call, the test would still work since it just needs ID to be functional

Reply:
Fair. Should we drop this test then?

Reply:
I think the test is valuable, just need to change it to `compositeKeysTable` and have 2 records with the same partition key and different sort keys. We can then delete using `Expression` on 1 of the sort key and the expected result should be the other one.